### PR TITLE
Remove the page `Referrers`

### LINKED ISSUE #27064: Remove the page "Referrers" after the upgrade
**Describe**
Since the module  linked to this page are archived, the page has no more use.
**Solution**
- Remove the page at the upgrade

- When the core removes a page, we should remove the quick access linked to this page.

- Remove SEO & URLs > Referrers for the BO
and on the user doc when the PR will be merged
remove the label

_Originally posted by @MatShir in https://github.com/PrestaShop/PrestaShop/issues/26662#issuecomment-999455634_

### PR DESCRIPTION
**Pull request by Progi1984**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Remove the page `Referrers`
| Type?             | refacto
| Category?         | BO
| BC breaks?        | yes
| Deprecations?     | no
| Fixed ticket?     | Fixes #27064
| How to test?      | Cf. #27064

**:notebook: BC Breaks**
* Removed template
  * `admin-dev/themes/default/template/controllers/referrers/calendar.tpl`
  * `admin-dev/themes/default/template/controllers/referrers/form_settings.tpl`
  * `admin-dev/themes/default/template/controllers/referrers/helpers/form/form.tpl`
  * `admin-dev/themes/default/template/controllers/referrers/helpers/form/index.tpl`
  * `admin-dev/themes/default/template/controllers/referrers/helpers/list/list_header.tpl`
  * `admin-dev/themes/default/template/controllers/referrers/helpers/view/view.tpl`
* Removed class `Referrer` 
* Removed controller class `AdminReferrersControllerCore` 
* Removed class `PrestaShop\PrestaShop\Core\Domain\Customer\QueryResult\ReferrerInformation` 
* Removed configuration `TRACKING_DIRECT_TRAFFIC`
* Removed parameter `$referrersInformation` from constructor of the class `PrestaShop\PrestaShop\Core\Domain\Customer\QueryResult\ViewableCustomer`

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/27098)

### CODE REVIEW

**Review by matthieu-rolland — DISMISSED** [review]

**Review by atomiix — DISMISSED** [review]

**Comment by khouloudbelguith** [comment]
Hi @Progi1984,

I have an issue following these steps:
1. Go to BO > Advanced Parameters > Team > Employees page
2. Create an employee with a `Salesman` profile 
image
3. Try to sign in with a new employee created
4. Go to BO > Shop Parameters > Traffic & Seo
5. See error 

image


I checked the Salesman profile on the permissions page, this page is not displayed
image


Thanks!

**Comment by Progi1984** [comment]
@matthieu-rolland @atomiix I fixed the bug found by @khouloudbelguith

**Review by atomiix — DISMISSED** [review]

**Review by matthieu-rolland — COMMENTED** [review]
this PR needs to be rebased

**Comment by Progi1984** [comment]
@matthieu-rolland & @atomiix Rebased.

**Review by atomiix — DISMISSED** [review]

File: classes/ConnectionsSource.php
Comment (jolelievre) [inline classes/ConnectionsSource.php:54]: I would naively think that if the Referrer feature is removed then this whole method becomes useless? Unless it's badly named and is used for other purposes?

But at first sight, I'm thinking: if don't handle Referrer anymore then we never need to log them no?

File: install-dev/data/db_structure.sql
Comment (jolelievre) [inline install-dev/data/db_structure.sql:1912]: You need to update the SQL scripts in the autoupgrade module right? It's better to do it now or I'm sure we'll forget about it ^^

**Review by jolelievre — COMMENTED** [review]
Thanks @Progi1984 

I just have one question before approval.
And the autoupgrade module needs to be updated for SQL changes.

Does this need QA approval? or can it be merged if CI is green?

File: install-dev/data/db_structure.sql
Comment (Progi1984) [reply install-dev/data/db_structure.sql:1912]: Thanks. It's done : PrestaShop/autoupgrade#447

File: classes/ConnectionsSource.php
Comment (Progi1984) [reply classes/ConnectionsSource.php:54]: It's badly named. It allows to log connection source (and before this PR, it logged referrer to this connection source).

**Comment by Progi1984** [comment]
@jolelievre It should be checked by the @PrestaShop/qa-functional team.

**Review by jolelievre — DISMISSED** [review]
Thanks @Progi1984

**Review by PierreRambaud — DISMISSED** [review]

**Comment by Progi1984** [comment]
@PierreRambaud @jolelievre @atomiix Rebased.

**Review by jolelievre — COMMENTED** [review]
Thanks @Progi1984 

All good for me, however I didn't approve because another conflict arose Waiting fro new rebase to approve it Don't hesitate to ping me on direct message to validate quickly after rebase

Also, I think this would need a PR in auto-upgrade to clean all the elements you removed here from the Database

**Comment by Progi1984** [comment]
@jolelievre It's rebased ;)

**Review by jolelievre — APPROVED** [review]
Thanks @Progi1984

**Review by atomiix — APPROVED** [review]

**Comment by Progi1984** [comment]
@jolelievre It's already done : PrestaShop/autoupgrade/pull/447

**Review by PierreRambaud — APPROVED** [review]

**Review by khouloudbelguith — CHANGES_REQUESTED** [review]
Hi @Progi1984,

After an upgrade from PS1778 to this PR
image
In the BO > Shop Parameters > SEO & URL page
The Referrers Tab is still displayed

image
and an exception is displayed in this page
image

Thanks!

**Comment by Progi1984** [comment]
@khouloudbelguith It's a fix to do in autoupgrade. Not in the core. But I will make a PR in `autoupgrade` tomorrow ;)

**Comment by Progi1984** [comment]
@khouloudbelguith I create a PR PrestaShop/autoupgrade/pull/466 for fixing this issue. (No changes on the Core so I add a "Waiting for QA").

**Review by khouloudbelguith — CHANGES_REQUESTED** [review]
Hi @Progi1984,

I have an issue following these steps:
1. Install this PR
2. Go to BO > International > Translations Page
3. Type of translation = BO & select a language and click on Modify
4. Search `Referrers`
5. You can find 3 results that match the query "Referrers".
5.1. `Referrers/affiliates` in `Admin > Advparameters > Feature` => I don't know if it is normal behavior or a bug
5.2. `Referrers` in `Admin > Navigation > Menu` => I think it is a bug because we are removed this page from the menu 
image
5.3. `Referrers` in `Admin > Orderscustomers > Feature` => I don't know if it is normal behavior or a bug

Thanks!

**Comment by Progi1984** [comment]
As discussed with @khouloudbelguith, translations for v8.0.0 have not been cleaning. They will be during the Wording Freeze. (ping @Julievrz).

**Review by khouloudbelguith — APPROVED** [review]
Hi @Progi1984,

It is ok :heavy_check_mark: 
1. I checked the search from BO
image
2. I checked the BDD => Ok
image
3. I checked the permission page
image
4. I checked with multistore page
image
5. I checked a mobile
image

6. I tried to create another employee with `superadmin` profile => ok
7. I tried to create other employees with different profiles (SEO & URLs accessible and not accessible) => OK

Thanks!

**Comment by Progi1984** [comment]
Thanks @khouloudbelguith

**Pull request merged**
